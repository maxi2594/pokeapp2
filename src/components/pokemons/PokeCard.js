import React from 'react'
import { Link } from 'react-router-dom';
import { useFetchPokes } from '../../hooks/useFetchPokes';

export const PokeCard = ({ pokemon, sprites }) => {
    
    const styles = {
        cardCenter: {
            alignItems: 'center',
            display: 'flex',
            justifyContent: 'center',
            backgroundColor: 'rgb(63,67,71)',
            borderRadius: '100px'
        },
        cardStyles: {
            alignItems: 'center',
            display: 'flex',
            justifyContent: 'center',
            border: '2px solid grey',
            backgroundColor: '#FAFAFA',
            maxWidth: '400px',
            margin: '5px',
            padding: '10px'
        },
        cardImg: {
            width: '150px',
            height: '150px',
        }
    }

    return (
        
        <div className="card col-sm-6 mb-2 animate__animated animate__fadeIn animate__delay-.5s" style={ styles.cardStyles} >
            <div className="col-md-6">
                <div style={styles.cardCenter}>

                    <img
                        style={ styles.cardImg}
                        src={ sprites.front_default } 
                        className="card-img"
                        alt={ pokemon.name }/>

                </div>
            </div>
            <div className="col-md-8">
                <div className="card-body">

                    <h2
                        style={{textAlign: 'center'}}
                    > { pokemon.name } 
                    </h2>

                    <hr></hr>    
                </div>
            </div>
        </div>
)
}

