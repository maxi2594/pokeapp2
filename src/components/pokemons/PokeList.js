import React from 'react';
import PropTypes from "prop-types";

import { PokeCard } from './pokeCard/PokeCard';


export const PokeList = ({results}) => {
    
    const styles = {
        display: 'flex',
        flexWrap: 'wrap',
        flexDirection: 'row',
        justifyContent: 'space-around',
    }
    return (
            <ul style={ styles }>


                {
                    results &&
                    (results.map( character => (
                        <PokeCard 
                            {...character }
                            key={ character.name }
                            />
                    )))
                }
            </ul>
    )
}

PokeList.propTypes = {
    results: PropTypes.array.isRequired
}