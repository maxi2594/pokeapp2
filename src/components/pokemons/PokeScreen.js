import React, { useCallback, useContext, useEffect, useState } from 'react'
import { pokeContext } from '../../context/pokeContext'
import { getCharactersData } from '../../helpers/getPokesApi'
import { NavigateButtons } from '../UI/buttons/NavigateButtons'
import { LoadIcon } from '../UI/loading/LoadIcon'
import { PokeList } from './PokeList'


export const PokeScreen = () => {

    const styles = {
        display: 'flex',
        alignItems : 'center',
        backgroundColor: 'rgb(43,47,51)',
        justifyContent: 'center',
        flexDirection: 'column',
        marginTop: '2rem',
        width: '100%',
        height: '100%'
    }

    const {state, handleSetUrL, stopLoading, startLoading } = useContext(pokeContext);
    const { data, loading, error} = state;
    
    const {results, next, previous } = data

    const [characters, setstate] = useState([])

    const getData = useCallback(
        async () => {
            if (results) {
                
                const data =  await getCharactersData(results)
                return data
            }
                            
        },
        [results],
    );

    useEffect(() => {
        startLoading()
        getData().then( data => {

            setTimeout(() => {
                
                setstate( data )
                stopLoading();
            }, 1000);
            
        })
    }, [getData])

    return (
        <div style={ styles }>
            <NavigateButtons 
                        handleSetUrL={ handleSetUrL }
                        next={ next }
                        previous={ previous }
            />
            { (loading) ? 
                (<LoadIcon/>)
                :
                (<>
                    
                    <PokeList results={characters} />
                    
                </>
                )
            }
            <NavigateButtons 
                        handleSetUrL={ handleSetUrL }
                        next={ next }
                        previous={ previous }
            />
            { 
                (error) && (alert( error ))
            }
        </div>
    )
}
