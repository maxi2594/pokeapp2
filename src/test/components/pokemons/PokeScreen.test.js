import { mount, shallow } from 'enzyme'
import React from 'react'
import { MemoryRouter } from 'react-router'
import { PokeScreen } from '../../../components/pokemons/PokeScreen'
import { pokeContext } from '../../../context/pokeContext'

describe('test in <PokeScreen/>', () => {

    const startLoading = jest.fn();
    const stopLoading = jest.fn();
    const getData = jest.fn();


    const componentProps = {
        state: {
            data: {
                results: [], 
                next: 'next', 
                previous: 'previous'
            }, 
            loading: true, 
            error: null
        },
        url: '',
        handleSetUrL: jest.fn(),
        stopLoading: jest.fn(),
        startLoading:jest.fn() 
    } 

    const wrapperFn = (props) => {
        return (
            mount(
                <pokeContext.Provider value={{...props}}>
                    <MemoryRouter>
                        <PokeScreen/>
                        </MemoryRouter>
                </pokeContext.Provider>
            )
        )
    }

    test('should be shown properly', () => {
        const wrapper = wrapperFn(componentProps)
        expect(wrapper).toMatchSnapshot()
    })

    test('should contain 2 NavigateButtons & LoadIcon', () => {
        const wrapper = wrapperFn(componentProps)
        expect(wrapper.find('NavigateButtons').length).toEqual(2)
        expect(wrapper.find('LoadIcon').exists()).toBe(true)
    })

    test('should contain PokeList', () => {
        const componentProps2 = {
            ...componentProps,
            state: {
                data: {
                    results: [], 
                    next: 'next', 
                    previous: 'previous'
                }, 
                loading: false, 
                error: null
            }
        }
        // console.log(componentProps2);
        const wrapper = wrapperFn(componentProps2)

        expect(wrapper.find('PokeList').exists()).toBe(true)       
    })
    
})
