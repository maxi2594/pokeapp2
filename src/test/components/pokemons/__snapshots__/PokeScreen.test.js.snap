// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <PokeScreen/> should be shown properly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(43,47,51)",
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "marginTop": "2rem",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(43,47,51)",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": "2rem",
        "marginTop": "1rem",
      }
    }
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "2px solid grey",
          "color": "grey",
          "height": "2rem",
          "marginLeft": "10px",
          "marginRight": "10px",
          "width": "5rem",
        }
      }
    >
      PREVIOS
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "2px solid grey",
          "color": "grey",
          "height": "2rem",
          "marginLeft": "10px",
          "marginRight": "10px",
          "width": "5rem",
        }
      }
    >
      NEXT
    </button>
  </div>
  <div
    className="lds-dual-ring"
  />
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(43,47,51)",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": "2rem",
        "marginTop": "1rem",
      }
    }
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "2px solid grey",
          "color": "grey",
          "height": "2rem",
          "marginLeft": "10px",
          "marginRight": "10px",
          "width": "5rem",
        }
      }
    >
      PREVIOS
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "2px solid grey",
          "color": "grey",
          "height": "2rem",
          "marginLeft": "10px",
          "marginRight": "10px",
          "width": "5rem",
        }
      }
    >
      NEXT
    </button>
  </div>
</div>
`;
